### PR TITLE
Support searching by attribute name in data dictionary

### PR DESCRIPTION
**Pull request by roadlittledawn**
## Description

Replaces the filter for selecting a specific attribute with a text input to search for attributes by name. Just matches on any instance of the search string in the attribute name. It lower cases the search string and the attribute names when matching.

Adds ability to apply filters by typing `enter`.

## Related issues / PRs

Closes https://github.com/newrelic/docs-website/issues/626

## Screenshot(s)

https://user-images.githubusercontent.com/2952843/125660278-2a5aed92-3364-4001-802c-9507f4e3255a.mp4

### CODE REVIEW

**Comment by roadlittledawn** [comment]
i added a spinner while it's performing a search, buuut not 100% sure on the implementation because i had to use a `setTimeout()` in `useEffect()`. not sure that's ideal because it may prolong some searches a bit. i do like some visual cue that something is happening though since it sometimes takes ~3s to search (esp when rendering most of the events)

**Comment by roadlittledawn** [comment]
looks like there's some prettier reformatting in the `attribute-dictionary.js` file ¯\_(ツ)_/¯

File: src/pages/attribute-dictionary.js [hunk 1]
```diff
@@ -180,207 +170,219 @@ AttributeDictionary.propTypes = {
...
-        <div
+        <h2
           css={css`
-            position: relative;
+            position: sticky;
+            top: var(--global-header-height);
+            background: var(--primary-background-color);
+            padding: 1rem 0;
+
+            // cover up the right table border
```
Comment (zstix) [inline src/pages/attribute-dictionary.js:206]: Oh neat, I didn't realize we could do comments like this!

**Review by zstix — APPROVED** [review]
Nice work!!!

**Comment by paperclypse** [comment]
YES! So cool!

**Pull request merged**